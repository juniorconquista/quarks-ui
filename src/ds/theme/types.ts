import {
    ColorProps,
    FlexboxProps,
    LayoutProps,
    SpaceProps,
    PositionProps,
    BorderProps,
    GridProps,
} from 'styled-system';

export type DefaultStyleProps = BorderProps &
    ColorProps &
    FlexboxProps &
    LayoutProps &
    PositionProps &
    SpaceProps &
    GridProps & {
        // for some reason this isn't included; this is for replacing the 'base' component
        // as?: React.ComponentType;
    };
