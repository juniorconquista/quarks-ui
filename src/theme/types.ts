import {
    ColorProps,
    FlexboxProps,
    LayoutProps,
    SpaceProps,
    PositionProps,
    BorderProps,
    GridProps,
} from 'styled-system';

export type CssProperties = {
    transition?: string;
    filter?: string;
    textTransform?:
        | 'none'
        | 'capitalize'
        | 'uppercase'
        | 'lowercase'
        | 'initial'
        | 'inherit';
    animationFillMode?:
        | 'none'
        | 'forwards'
        | 'backwards'
        | 'both'
        | 'initial'
        | 'inherit';
    objectFit?: 'contain' | 'cover' | 'fill' | 'none' | 'scale-down';
    whiteSpace?:
        | 'none'
        | 'nowrap'
        | 'pre'
        | 'pre-line'
        | 'pre-wrap'
        | 'initial'
        | 'inherit';
    '-webkit-box-shadow'?: string;
};

export type DefaultStyleProps = BorderProps &
    ColorProps &
    FlexboxProps &
    LayoutProps &
    PositionProps &
    SpaceProps &
    GridProps & {
        // for some reason this isn't included; this is for replacing the 'base' component
        // as?: React.ComponentType;
    };
