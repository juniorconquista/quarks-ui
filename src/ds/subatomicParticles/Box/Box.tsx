import { SFC } from 'react';
import styled from 'styled-components';
import {
    variant,
    VariantArgs,
    color,
    typography,
    fontSize,
    position,
    space,
    flexbox,
    grid,
    layout,
    border,
    system,
} from 'styled-system';
import { BoxProps } from './box.types';

const variants = variant({
    prop: 'styling',
    variants: {
        base: {
            display: 'flex',
            flexDirection: 'column',
            justifyContent: 'center',
            alignItems: 'center',
            '&:focus': {
                outline: 'none',
            },
        },
        get row() {
            return {
                ...this.base,
                flexDirection: 'row',
            };
        },
        get column() {
            return {
                ...this.base,
                flexDirection: 'column',
            };
        },
        get grid() {
            return {
                ...this.base,
                display: 'grid',
                alignItems: 'start',
            };
        },
        overlay: {
            position: 'fixed',
            left: 0,
            right: 0,
            top: 0,
            bottom: 0,
            zIndex: 9999,
        },
    },
} as VariantArgs);

export const Box: SFC<BoxProps> = styled.div<BoxProps>`
    ${variants}
    ${color}
    ${typography}
    ${fontSize}
    ${space}
    ${position}
    ${flexbox}
    ${grid}
    ${layout}
    ${border}
    ${system({ transition: true, animationFillMode: true })}
    ${(props) => props.css ?? ''}
`;

export default Box;
