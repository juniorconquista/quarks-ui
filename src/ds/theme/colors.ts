export default {
    dark: false,
    primary: {
        default: '#F26522',
        dark: '#d14b0c',
        light: '#f69263',
        contrast: '#ffffff',
    },
    secondary: {
        default: '#372E2A',
        dark: '#2c2522',
        light: '#453935',
        contrast: '#1B2B3A',
    },
    danger: {
        default: '#ff6961',
        dark: '#ff261b',
        light: '#ffbdb9',
        contrast: '#ffffff',
    },
    success: {
        default: '#0E9d51',
        dark: '#0a753c',
        light: '#10bc60',
        contrast: '#ffffff',
    },
    warning: {
        default: '#ffb143',
        dark: '#e4902c',
        light: '#ffd868',
        contrast: '#ffffff',
    },
    info: {
        default: '#0ea7a7',
        dark: '#0a7676',
        light: '#11d4d4',
        contrast: '#ffffff',
    },
    grays: {
        default: '#9ba8b2',
        dark: '#4b4b4b',
        light: '#eeeeef',
        contrast: '#f5f5f5',
    },
    background: {
        default: '#ffffff',
        dark: '#F26522',
    },
    text: {
        light: '#ffffff',
        dark: '#1B2b3a',
    },
    disabled: '#9398a6',
    primaries: ['#d7edf4'],
    whites: ['#eeeeef', '#f2fbfe'],
    blacks: ['#35363a'],
    hovers: ['#f5f7f7'],
};
