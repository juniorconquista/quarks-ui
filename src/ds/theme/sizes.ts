export const breakpoints = ['0rem', '37.5rem', '62rem'];

export const space = {
    none: '0',
    xxxs: '0.285rem',
    xxs: '0.571rem',
    xs: '0.857rem',
    sm: '1.142rem',
    md: '1.714rem',
    lg: '2.5rem',
    xl: '3.428rem',
    xxl: '6.857rem',
};

export const fontSizes = [
    '0.5rem',
    '0.85rem',
    '1rem',
    '1.5rem',
    '1.8rem',
    '2rem',
];

export const buttonSizes = {
    medium: {
        paddingTop: 8,
        paddingBottom: 8,
        paddingRight: 16,
        paddingLeft: 16,
        borderRadius: 16,
    },
    small: {
        paddingTop: 4,
        paddingBottom: 4,
        paddingRight: 16,
        paddingLeft: 16,
        borderRadius: 16,
    },
};

export const radii = {
    none: '0',
    sm: '0.285rem',
    md: '2.25rem',
    lg: '1.142rem',
    xl: '1.714rem',
};
