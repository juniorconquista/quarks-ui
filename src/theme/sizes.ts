export const breakpoints = ['0rem', '37.5rem', '62rem']

export const space = {
  none: '0',
  xxxs: '0.285rem',
  xxs: '0.571rem',
  xs: '0.857rem',
  sm: '1.142rem',
  md: '1.714rem',
  lg: '2.5rem',
  xl: '3.428rem',
  xxl: '6.857rem'
}

export const fontSizes = [
  '0.5rem', // 8px
  '0.67rem', // 10.72px - H6 / SMALL
  '0.83rem', // 13.28px - H5
  '0.875rem', // 14px
  '1rem', // 16px - H4/P/SPAN
  '1.17rem', // 18.72px - H3
  '1.5rem', // 24px - H2
  '1.8rem', // 28.8px
  '2rem' // 32px - H1
]

export const buttonSizes = {
  medium: {
    paddingTop: 8,
    paddingBottom: 8,
    paddingRight: 16,
    paddingLeft: 16,
    borderRadius: 16
  },
  small: {
    paddingTop: 4,
    paddingBottom: 4,
    paddingRight: 16,
    paddingLeft: 16,
    borderRadius: 16
  }
}

export const radii = {
  none: '0',
  sm: '0.285rem',
  md: '2.25rem',
  lg: '1.142rem',
  xl: '1.714rem'
}
