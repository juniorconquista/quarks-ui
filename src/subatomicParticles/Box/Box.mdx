---
name: Box
menu: Layout
---

import { Playground, Props } from 'docz';
import Box from './Box';

# Box

Search engines use the headings to index the structure and content of your web pages. Users often skim a page by its headings. It is important to use headings to show the document structure.

Each HTML heading has a default size. However, you can specify the size for any heading with the style attribute, using the CSS _font-size_ property.

### Best practices

-   Use HTML headings for headings only. Don't use headings to make text BIG or bold.

## Properties

<Props of={Box} />

## Basic usage

```javascript
import { Box } from 'quarks-ui/subatomicParticles';
```

<Playground>
    <Box styling="row">Row</Box>
</Playground>
