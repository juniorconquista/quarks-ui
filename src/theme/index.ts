import colors from './colors'
import { breakpoints, space, fontSizes, radii } from './sizes'

const theme = {
  fontSizes,
  space,
  breakpoints,
  radii,
  colors: colors()
}

// @ts-expect-error
theme.breakpoints.mobile = breakpoints[0]
// @ts-expect-error
theme.breakpoints.tablet = breakpoints[1]
// @ts-expect-error
theme.breakpoints.desktop = breakpoints[2]

export default theme
