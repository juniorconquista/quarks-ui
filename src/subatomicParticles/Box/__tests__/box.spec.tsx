import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import 'jest-styled-components';

import Box from '../Box';
import theme from '../../../theme';

describe('Box', () => {
    test('basic snapshot', () => {
        const tree = renderer
            .create(
                <ThemeProvider theme={theme}>
                    <Box styling="base" />
                </ThemeProvider>,
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('should inherit the styling properties of the base variant', () => {
        const tree = renderer
            .create(
                <ThemeProvider theme={theme}>
                    <Box styling="base" />
                </ThemeProvider>,
            )
            .toJSON();
        expect(tree).toHaveStyleRule('display', 'flex');
        expect(tree).toHaveStyleRule('flex-direction', 'column');
        expect(tree).toHaveStyleRule('justify-content', 'center');
        expect(tree).toHaveStyleRule('align-items', 'center');
        expect(tree).toHaveStyleRule('outline', 'none', {
            modifier: '&:focus',
        });
    });

    test('should inherit the styling properties of the row variant', () => {
        const tree = renderer
            .create(
                <ThemeProvider theme={theme}>
                    <Box styling="row" />
                </ThemeProvider>,
            )
            .toJSON();
        expect(tree).toHaveStyleRule('display', 'flex');
        expect(tree).toHaveStyleRule('flex-direction', 'row');
        expect(tree).toHaveStyleRule('justify-content', 'center');
        expect(tree).toHaveStyleRule('align-items', 'center');
        expect(tree).toHaveStyleRule('outline', 'none', {
            modifier: '&:focus',
        });
    });

    test('should inherit the styling properties of the column variant', () => {
        const tree = renderer
            .create(
                <ThemeProvider theme={theme}>
                    <Box styling="column" />
                </ThemeProvider>,
            )
            .toJSON();
        expect(tree).toHaveStyleRule('display', 'flex');
        expect(tree).toHaveStyleRule('flex-direction', 'column');
        expect(tree).toHaveStyleRule('justify-content', 'center');
        expect(tree).toHaveStyleRule('align-items', 'center');
        expect(tree).toHaveStyleRule('outline', 'none', {
            modifier: '&:focus',
        });
    });

    test('should inherit the styling properties of the grid variant', () => {
        const tree = renderer
            .create(
                <ThemeProvider theme={theme}>
                    <Box styling="grid" />
                </ThemeProvider>,
            )
            .toJSON();
        expect(tree).toHaveStyleRule('display', 'grid');
        expect(tree).toHaveStyleRule('flex-direction', 'column');
        expect(tree).toHaveStyleRule('justify-content', 'center');
        expect(tree).toHaveStyleRule('align-items', 'start');
        expect(tree).toHaveStyleRule('outline', 'none', {
            modifier: '&:focus',
        });
    });

    test('should inherit the styling properties of the overlay variant', () => {
        const tree = renderer
            .create(
                <ThemeProvider theme={theme}>
                    <Box styling="overlay" />
                </ThemeProvider>,
            )
            .toJSON();
        expect(tree).toHaveStyleRule('position', 'fixed');
        expect(tree).toHaveStyleRule('left', '0');
        expect(tree).toHaveStyleRule('right', '0');
        expect(tree).toHaveStyleRule('top', '0');
        expect(tree).toHaveStyleRule('bottom', '0');
        expect(tree).toHaveStyleRule('z-index', '9999');
    });

    test('should apply custom border styling properties', () => {
        const tree = renderer
            .create(<Box styling="row" border="1px solid" />)
            .toJSON();
        expect(tree).toHaveStyleRule('border', '1px solid');
        expect(tree).toHaveStyleRule('display', 'flex');
        expect(tree).toHaveStyleRule('flex-direction', 'row');
        expect(tree).toHaveStyleRule('justify-content', 'center');
        expect(tree).toHaveStyleRule('align-items', 'center');
        expect(tree).toHaveStyleRule('outline', 'none', {
            modifier: '&:focus',
        });
    });

    test('should apply custom color styling properties', () => {
        const tree = renderer
            .create(<Box styling="row" border="1px solid" color="blue" />)
            .toJSON();
        expect(tree).toHaveStyleRule('color', 'blue');
        expect(tree).toHaveStyleRule('border', '1px solid');
        expect(tree).toHaveStyleRule('display', 'flex');
        expect(tree).toHaveStyleRule('flex-direction', 'row');
        expect(tree).toHaveStyleRule('justify-content', 'center');
        expect(tree).toHaveStyleRule('align-items', 'center');
        expect(tree).toHaveStyleRule('outline', 'none', {
            modifier: '&:focus',
        });
    });

    test('should apply custom layout styling properties', () => {
        const tree = renderer
            .create(<Box styling="row" border="1px solid" width={1 / 2} />)
            .toJSON();
        expect(tree).toHaveStyleRule('width', '50%');
        expect(tree).toHaveStyleRule('border', '1px solid');
        expect(tree).toHaveStyleRule('display', 'flex');
        expect(tree).toHaveStyleRule('flex-direction', 'row');
        expect(tree).toHaveStyleRule('justify-content', 'center');
        expect(tree).toHaveStyleRule('align-items', 'center');
        expect(tree).toHaveStyleRule('outline', 'none', {
            modifier: '&:focus',
        });
    });

    test('should apply custom position styling properties', () => {
        const tree = renderer
            .create(<Box styling="row" position="absolute" />)
            .toJSON();
        expect(tree).toHaveStyleRule('position', 'absolute');
        expect(tree).toHaveStyleRule('display', 'flex');
        expect(tree).toHaveStyleRule('flex-direction', 'row');
        expect(tree).toHaveStyleRule('justify-content', 'center');
        expect(tree).toHaveStyleRule('align-items', 'center');
        expect(tree).toHaveStyleRule('outline', 'none', {
            modifier: '&:focus',
        });
    });

    test('should apply custom position styling properties', () => {
        const tree = renderer.create(<Box styling="row" m={2} />).toJSON();
        expect(tree).toHaveStyleRule('margin', '8px');
        expect(tree).toHaveStyleRule('display', 'flex');
        expect(tree).toHaveStyleRule('flex-direction', 'row');
        expect(tree).toHaveStyleRule('justify-content', 'center');
        expect(tree).toHaveStyleRule('align-items', 'center');
        expect(tree).toHaveStyleRule('outline', 'none', {
            modifier: '&:focus',
        });
    });

    test('should apply custom typography styling properties', () => {
        const tree = renderer
            .create(<Box styling="base" fontSize={8} />)
            .toJSON();
        expect(tree).toHaveStyleRule('font-size', '72px');
        expect(tree).toHaveStyleRule('display', 'flex');
        expect(tree).toHaveStyleRule('flex-direction', 'column');
        expect(tree).toHaveStyleRule('justify-content', 'center');
        expect(tree).toHaveStyleRule('align-items', 'center');
        expect(tree).toHaveStyleRule('outline', 'none', {
            modifier: '&:focus',
        });
    });
});
