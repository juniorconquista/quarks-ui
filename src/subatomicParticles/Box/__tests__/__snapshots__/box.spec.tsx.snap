// Jest Snapshot v1, https://goo.gl/fbAQLP

exports['Box basic snapshot 1'] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0:focus {
  outline: none;
}

<div
  className="c0"
/>
`
